pkg/kubernetes/version.go

```diff
@@ -0,0 +1,220 @@
+package kubernetes
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
+	AnnotationSpinnakerArtifactVersion = `artifact.spinnaker.io/version`
+	AnnotationSpinnakerMonikerSequence = `moniker.spinnaker.io/sequence`
+	Annotation
+)
+
+type SpinnakerVersion struct {
+	Long  string
+	Short string
+}
+
+//Create a similar function for labels
+func (c *controller) AddSpinnakerVersionAnnotations(u *unstructured.Unstructured, version SpinnakerVersion) error {
+	annotate(u, AnnotationSpinnakerArtifactVersion, version.Long)
+	annotate(u, AnnotationSpinnakerMonikerSequence, version.Short)
+
+	var err error
+
+	gvk := u.GroupVersionKind()
+
+	if strings.EqualFold(gvk.Kind, "deployment") {
+		d := NewDeployment(u.Object)
+
+		d.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		d.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = d.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(gvk.Kind, "replicaset") {
+		rs := NewReplicaSet(u.Object)
+
+		rs.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		rs.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = rs.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	if strings.EqualFold(gvk.Kind, "daemonset") {
+		ds := NewReplicaSet(u.Object)
+
+		ds.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+		ds.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+		*u, err = ds.ToUnstructured()
+		if err != nil {
+			return err
+		}
+	}
+
+	//Todo: create statefulset.go with all the needed functions
+	// if strings.EqualFold(gvk.Kind, "statefulset") {
+	// 	ss := NewStatefulSet(u.Object)
+
+	// 	ss.AnnotateTemplate(AnnotationSpinnakerArtifactVersion, version.Long)
+	// 	ss.AnnotateTemplate(AnnotationSpinnakerMonikerSequence, version.Short)
+
+	// 	*u, err = ds.ToUnstructured()
+	// 	if err != nil {
+	// 		return err
+	// 	}
+	// }
```

Comment:
Has this been created? Can we remove these comments?

Reply:
It [looks like it has](https://github.com/homedepot/go-clouddriver/blob/master/pkg/kubernetes/statefulset.go#L22). Let's add implementation for this.